forui/test/src/widgets/select_group/select_group_controller_test.dart

```diff
@@ -0,0 +1,118 @@
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:forui/forui.dart';
+
+void main() {
+  group('FRadioSelectGroupController', () {
+    test('should initialize with a single value', () {
+      final controller = FRadioSelectGroupController(value: 1);
+
+      expect(controller.values, equals({1}));
+    });
+
+    test('should initialize with an empty set when no value is provided', () {
+      final controller = FRadioSelectGroupController<int>();
+
+      expect(controller.values, isEmpty);
+    });
+
+    test('should change selection when a new value is selected', () {
+      final controller = FRadioSelectGroupController<int>()..onChange(1, true);
+
+      expect(controller.values, equals({1}));
+    });
+
+    test('should not change selection when the same value is selected', () {
+      final controller = FRadioSelectGroupController(value: 1)..onChange(1, true);
+
+      expect(controller.values, equals({1}));
+    });
+
+    test('should not change selection when trying to deselect', () {
+      final controller = FRadioSelectGroupController(value: 1)..onChange(1, false);
+
+      expect(controller.values, equals({1}));
+    });
+
+    test('contains(...)', () {
+      final controller = FRadioSelectGroupController(value: 1);
+
+      expect(controller.contains(1), isTrue);
+      expect(controller.contains(2), isFalse);
+
+      controller.onChange(2, true);
+
+      expect(controller.contains(1), isFalse);
+      expect(controller.contains(2), isTrue);
+    });
+
+    test('ensure notifyListeners() is called', () {
+      bool notified = false;
+
+      FRadioSelectGroupController<int>()
+        ..addListener(() => notified = true)
+        ..onChange(1, true);
+
+      expect(notified, isTrue);
```

Comment:
Should we merge this test with the other tests above? We'll able to test whether notifyListener() should be called in those cases.